### PR TITLE
EWPP-1901: Update registration button information at the exact time.

### PR DESCRIPTION
**Pull request by sergepavle**
### Description

[Insert description here]

### Change log

- Added:
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: modules/oe_theme_content_event/js/event_registration.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
+/**
+ * @file
+ * Attaches behaviors for Event registration.
+ */
+(function (Drupal, drupalSettings) {
+  /**
+   * Shows related information when registration is active.
```
Comment (imanoleguskiza) [inline modules/oe_theme_content_event/js/event_registration.js]: Maybe something like: Shows registration information when the registration period is open.

File: modules/oe_theme_content_event/src/Plugin/ExtraField/Display/InfoDisclosureExtraFieldBase.php [hunk 2]
```diff
@@ -76,22 +76,13 @@ protected function isCurrentDay(int $timestamp): bool {
   }
```
Comment (imanoleguskiza) [inline modules/oe_theme_content_event/src/Plugin/ExtraField/Display/InfoDisclosureExtraFieldBase.php:77]: Can we improve the method description for isCurrentDay? Maybe something like: Asserts if a given timestamp corresponds to the same day of the current request time.

File: modules/oe_theme_content_event/src/Plugin/ExtraField/Display/OnlineDisplayExtraFieldBase.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_theme_content_event\Plugin\ExtraField\Display;
+
+/**
+ * Base class for fields that require display livestream link in exact time.
```
Comment (imanoleguskiza) [inline modules/oe_theme_content_event/src/Plugin/ExtraField/Display/OnlineDisplayExtraFieldBase.php]: Maybe: Base class for fields rendering livestream information.

File: modules/oe_theme_content_event/oe_theme_content_event.module [hunk 4]
```diff
@@ -21,6 +21,8 @@ function oe_theme_content_event_theme() {
         'label' => '',
         'url' => '',
         'description' => '',
+        'approaching_time_description' => '',
+        'approaching_time' => FALSE,
```
Comment (imanoleguskiza) [inline modules/oe_theme_content_event/oe_theme_content_event.module]: This name is a bit confusing. How about we call it "registration_day" and "registration_day_description"?

**Review by imanoleguskiza — DISMISSED** [review]

File: modules/oe_theme_content_event/tests/src/FunctionalJavascript/InfoDisclosureExtraFieldTest.php [hunk 5]
```diff
@@ -57,9 +57,12 @@ public function setUp(): void {
   }
 
   /**
-   * Tests that Event livestreaming link will be visible exactly on time.
+   * Tests that Event Online and Registration link are appeared exactly on time.
```
Comment (22Alexandra) [inline modules/oe_theme_content_event/tests/src/FunctionalJavascript/InfoDisclosureExtraFieldTest.php]:
```suggestion
   * Tests that Event Online and Registration links are displayed exactly on time.
```

File: modules/oe_theme_content_event/tests/src/FunctionalJavascript/InfoDisclosureExtraFieldTest.php [hunk 6]
```diff
@@ -92,12 +102,18 @@ public function testLivestreamDisplayDisclosing(): void {
       ]);
     $node->save();
     $this->drupalGet($node->toUrl());
+    // Check that livestream link is not prepared for disclosing.
     $livetime_elements = $this->getSession()->getPage()->findAll('css', '[data-livestream-element]');
```
Comment (22Alexandra) [inline modules/oe_theme_content_event/tests/src/FunctionalJavascript/InfoDisclosureExtraFieldTest.php]:
```suggestion
    $livestream_elements = $this->getSession()->getPage()->findAll('css', '[data-livestream-element]');
```

File: modules/oe_theme_content_event/tests/src/FunctionalJavascript/InfoDisclosureExtraFieldTest.php [hunk 7]
```diff
@@ -92,12 +102,18 @@ public function testLivestreamDisplayDisclosing(): void {
...
     $livestream_js = $this->xpath("//script[contains(@src, 'js/event_livestream.js')]");
     $this->assertCount(0, $livestream_js);
+    // Check that registration link is not prepared for disclosing.
+    $registration_elements = $this->getSession()->getPage()->findAll('css', '[data-registration-upcoming-element],[data-registration-active-element]');
+    $this->assertCount(0, $registration_elements);
+    $registration_js = $this->xpath("//script[contains(@src, 'js/event_registration.js')]");
+    $this->assertCount(0, $registration_js);
 
-    // Set livestream start date in 10 minutes later.
+    // Set start datetime in 10 minutes later.
```
Comment (22Alexandra) [inline modules/oe_theme_content_event/tests/src/FunctionalJavascript/InfoDisclosureExtraFieldTest.php]:
```suggestion
    // Set start datetime 10 minutes later.
```

**Review by 22Alexandra — CHANGES_REQUESTED** [review]

**Review by 22Alexandra — DISMISSED** [review]

**Review by 22Alexandra — APPROVED** [review]

**Review by upchuk — APPROVED** [review]
Blind for imanol

**Pull request merged**